src/Payum/Core/CoreGatewayFactory.php

```diff
@@ -71,13 +74,43 @@ public function createConfig(array $config = array())
         $config['twig.env'] = null;
 
         $config->defaults(array(
+            'httplug.message_factory'=>function(ArrayObject $config) {
+                if (class_exists(MessageFactoryDiscovery::class)) {
+                    return MessageFactoryDiscovery::find();
+                }
+
+                if (class_exists(\GuzzleHttp\Psr7\Request::class)) {
+                    return new GuzzleMessageFactory();
+                }
+
+                if (class_exists(\Zend\Diactoros\Request::class)) {
+                    return new DiactorosMessageFactory();
+                }
+
+                throw new \LogicException('The httplug.message_factory could not be guessed. Install one of the following packages: php-http/guzzle6-adapter, zendframework/zend-diactoros. You can also overwrite the config option with your implementation.');
+            },
+            'httplug.stream_factory'=>function(ArrayObject $config) {
+                if (class_exists(StreamFactoryDiscovery::class)) {
+                    return StreamFactoryDiscovery::find();
+                }
+
+                if (class_exists(\GuzzleHttp\Psr7\Request::class)) {
+                    return new GuzzleStreamFactory();
+                }
+
+                if (class_exists(\Zend\Diactoros\Request::class)) {
+                    return new DiactorosStreamFactory();
+                }
+
+                throw new \LogicException('The httplug.stream_factory could not be guessed. Install one of the following packages: php-http/guzzle6-adapter, zendframework/zend-diactoros. You can also overwrite the config option with your implementation.');
+            },
             'httplug.client'=>function(ArrayObject $config) {
                 if (class_exists(HttpClientDiscovery::class)) {
                     return HttpClientDiscovery::find();
                 }
 
                 if (class_exists(HttpGuzzle6Client::class)) {
-                    return new HttpGuzzle6Client();
+                    return HttpGuzzle6Client::createWithConfig(array());
```

Comment:
They already fixed this bug in 1.1.1, but Payum requires php-http/guzzle6-adapter ^1.0 in require-dev.
Without this change the tests are failing, so we need to update dependency or to leave this workaround